Seminars/Sem.03/Pract.03/Task01.cpp

```diff
@@ -1,102 +1,303 @@
+	exitProgram = 4
 };
 
 struct Student {
-    char name[max_length + 1];
-    char surname[max_length + 1];
-    char facultyNumber[max_digits + 1];
-    double averageGrade;
-    hairColor hair;
-};
+	char firstName[FIRST_NAME_LENGTH + 1];
+	char lastName[LAST_NAME_LENGTH + 1];
+	unsigned int facultyNumber;
+	double gpa;
+	HairColor hairColor;
+
+	Student() {}
 
-struct System {
-    Student students[max_users];
-    int studentsCount;
-
-    void addStudent(const Student &student) {
-        if (studentsCount >= max_users) {
-            cout << "Max users reached!" << endl;
-            return;
-        }
-        students[studentsCount] = student;
-        studentsCount++;
-    }
+	Student(const char* _firstName, const char* _lastName,
+		const unsigned int _facultyNumber, const double _gpa, const HairColor _hairColor) {
+		strcpy_s(firstName, _firstName);
+		strcpy_s(lastName, _lastName);
+		facultyNumber = _facultyNumber;
+		gpa = _gpa;
+		hairColor = _hairColor;
+	}
 };
 
-bool writeStudentsToFile(const char *filename, System &system) {
-    ofstream file(filename, std::ios::app);
-    if (!file.is_open()) {
-        cout << "File could not be opened!" << endl;
-        return false;
-    }
-    for (int i = 0; i < system.studentsCount; i++) {
-        file << system.students[i].name << "," << system.students[i].surname
-             << "," << system.students[i].facultyNumber << "," << system.students[i].averageGrade
-             << "," << hairColorNames[system.students[i].hair] << endl;
-    }
-    file.close();
-    return true;
+const char* enumFieldToString(const HairColor& hairColor) {
+	switch (hairColor)
+	{
+	case HairColor::brown:
+		return "brown";
+	case HairColor::black:
+		return "black";
+	case HairColor::blonde:
+		return "blonde";
+	case HairColor::red:
+		return "red";
+	case HairColor::white:
+		return "white";
+	default:
+		return "";
+	}
+}
+
+bool getMappedHairColor(const char* hairColorInput, HairColor& outputColor) {
+	if (strcmp(hairColorInput, "brown") == 0)
+	{
+		outputColor = HairColor::brown;
+		return true;
+	}
+	if (strcmp(hairColorInput, "black") == 0)
+	{
+		outputColor = HairColor::black;
+		return true;
+	}
+	if (strcmp(hairColorInput, "blonde") == 0)
+	{
+		outputColor = HairColor::blonde;
+		return true;
+	}
+	if (strcmp(hairColorInput, "red") == 0)
+	{
+		outputColor = HairColor::red;
+		return true;
+	}
+	if (strcmp(hairColorInput, "white") == 0)
+	{
+		outputColor = HairColor::white;
+		return true;
+	}
+
+	return false;
+}
+
+bool isNameInputValid(const char* str, const short MAX_STR_SIZE)
+{
+	return str[MAX_STR_SIZE] == '\0';
+}
+
+void initStudentFromConsole(Student& outputStudent) {
+	cin.ignore();
+
+	cout << "Input first name: ";
+
+	char firstName[FIRST_NAME_LENGTH + 1]{};
+	do
+	{
+		cin >> firstName;
+		if (!isNameInputValid(firstName, FIRST_NAME_LENGTH))
+		{
+			cout << "Please enter a valid name (max 16 chars): ";
+		}
+		else break;
+	} while (true);
+
+	cout << "Input last name: ";
```

Comment:
first and last name are the same code. Stick to the DRY and extract it to a function